### DISCUSSION TITLE
Using Historical Data Within Simulation

### DISCUSSION THREAD

**Discussion by ccastleb1** [john-harrold/ubiquity#20] [Q&A]
Hello! I was wondering if there was a way to use historical data within a simulation as it is running. I would like to set a dynamic secondary parameter to be a function of ODE states at prior time points, and I am having trouble doing that with the Ubiquity platform. Does anyone know of a way to call historical data in the system file?

Thank you for your help!

**Comment by john-harrold** [comment]
Howdy. It sounds like you want something like delayed differential equations. Is that what you're looking for?  

https://search.r-project.org/CRAN/refmans/deSolve/html/dede.html

If that's the case it's not currently supported in ubiquity. But the underlying solver (deSolve) does support it (see that link).  

Another way to interpret your question is that you are looking for a timevarying input that represents your state in the past. Something that could be read in from a dataset. You can do that with a time-varying covariate. 

https://r.ubiquity.tools/articles/Language.html#complicated-covariates

If that is what you want, I can create a simple example to show you how to do that.
